config/schemas/config.yaml

```diff
@@ -0,0 +1,2854 @@
+        properties:
+          persistence:
+            properties:
+              size:
+                default: 8Gi
+                type: string
+            type: object
+          resources:
+            $ref: '#/$defs/resources'
+        type: object
+    type: object
+  trivy:
+    properties:
+      affinity:
+        $ref: '#/$defs/affinity'
+      enabled:
+        default: true
+        type: boolean
+      resources:
+        $ref: '#/$defs/resources'
+      scanJobs:
+        properties:
+          concurrentLimit:
+            default: 1
+            type: number
+          retryDelay:
+            default: 1m
+            type: string
+          timeout:
+            default: 5m
+            type: string
+        type: object
+      serviceMonitor:
+        properties:
+          enabled:
+            default: true
+            type: boolean
+        type: object
+      tolerations:
+        $ref: '#/$defs/tolerations'
+      vulnerabilityScanner:
+        properties:
+          scanOnlyCurrentRevisions:
+            default: true
+            type: boolean
+          scannerReportTTL:
+            default: 720h
+            type: string
+        type: object
+    type: object
+  user:
+    properties:
+      alertmanager:
+        properties:
+          enabled:
+            default: true
+            type: boolean
+          ingress:
+            properties:
+              enabled:
+                default: false
+                type: boolean
+            type: object
+        type: object
+    type: object
+  velero:
+    properties:
+      enabled:
+        default: true
+        type: boolean
+      nodeAgent:
+        properties:
+          resources:
+            $ref: '#/$defs/resources'
+          tolerations:
+            $ref: '#/$defs/tolerations'
+        type: object
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      resources:
+        $ref: '#/$defs/resources'
+      retentionPeriod:
+        default: 720h0m0s
+        type: string
+      schedule:
+        default: 0 0 * * *
+        type: string
+      tolerations:
+        $ref: '#/$defs/tolerations'
+    type: object
+  welcomingDashboard:
+    # JSON Schema accepts additional undefined properties by default. To reject these instead:
+    additionalProperties: false
+    properties:
+      extraTextGrafana:
+        type: string
+      extraTextOpensearch:
+        type: string
+      extraVersions:
+        type: array
+        items:
+          type: object
+          properties:
+            name:
+              type: string
+            version:
+              type: string
+            subdomain:
+              type: string
+              format: hostname
+            url:
+              type: string
+              format: uri
+            releasenotes:
+              type: string
+              format: uri
+    type: object
+  elasticsearch: true
+  gatekeeper: true
+$defs:
+  $comment: Common types of things for re-use
+  affinity:
```

Comment:
Would it be possible to have these somewhere at the top?

Reply:
Done.

Could even be worth having a separate file for common definitions and then sc and wc specifics separate, depending on how much they differ.